src/PrestaShopBundle/Entity/Employee/Employee.php

```diff
@@ -0,0 +1,699 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Entity\Employee;
+
+use DateTime;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use PrestaShopBundle\Entity\Lang;
+use PrestaShopBundle\Security\Admin\SessionEmployeeInterface;
+use Symfony\Component\Security\Core\User\EquatableInterface;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @ORM\Entity(repositoryClass="PrestaShopBundle\Entity\Repository\EmployeeRepository")
+ *
+ * @ORM\Table(
+ *     indexes={
+ *
+ *         @ORM\Index(name="employee_login", columns={"email", "passwd"}),
+ *         @ORM\Index(name="id_employee_passwd", columns={"id_employee", "passwd"}),
+ *         @ORM\Index(name="id_profile", columns={"id_profile"}),
+ *     },
+ *  )
+ */
+class Employee implements UserInterface, PasswordAuthenticatedUserInterface, EquatableInterface, SessionEmployeeInterface
+{
+    public const ROLE_EMPLOYEE = 'ROLE_EMPLOYEE';
+
+    /**
+     * @ORM\Id
+     *
+     * @ORM\Column(name="id_employee", type="integer", options={"unsigned": true})
+     *
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private int $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="PrestaShopBundle\Entity\Employee\Profile")
+     *
+     * @ORM\JoinColumn(name="id_profile", referencedColumnName="id_profile", nullable=false, options={"unsigned": true})
+     */
+    private Profile $profile;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="PrestaShopBundle\Entity\Lang")
+     *
+     * @ORM\JoinColumn(name="id_lang", referencedColumnName="id_lang", nullable=false, options={"default": 0, "unsigned": true})
+     */
+    private Lang $defaultLanguage;
+
+    /**
+     * @var Collection<EmployeeSession>
+     *
+     * @ORM\OneToMany(targetEntity="PrestaShopBundle\Entity\Employee\EmployeeSession", mappedBy="employee", orphanRemoval=true, cascade={"persist"})
+     */
+    private Collection $sessions;
+
+    /**
+     * @ORM\Column(name="firstname", type="string")
+     */
+    private string $firstName;
+
+    /**
+     * @ORM\Column(name="lastname", type="string")
+     */
+    private string $lastName;
+
+    /**
+     * @ORM\Column(name="email", type="string")
+     */
+    private string $email;
```

Comment:
Is it interesting to add a unique constraint to this field?

Reply:
Argh my bad I misunderstood I thought you were talking about assertions for the entity validation Regarding the unicity of this column you're completely right But it's a change of the DB structure I prefer to handle it in another PR, we also discussed in another comment about removing the theme customizations field, I guess all this could be handled in a dedicated PR

Reply:
Ok :+1: